azurerm/internal/services/iotcentral/tests/resource_arm_iotcentral_application_test.go

```diff
@@ -0,0 +1,232 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+)
+
+func TestAccAzureRMIoTCentralApplication_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iotcentral_application", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotCentralApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotCentralApplication_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMIoTCentralApplication_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iotcentral_application", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotCentralApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotCentralApplication_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMIotCentralApplication_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotCentralApplicationExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMIoTCentralApplication_complete(t *testing.T) {
```

Comment:
move this function to ahead of update function

Reply:
@xuzhang3